Devoncarew sharing ui

devoncarew opened the pull request:
- create a dialog component
- create a glass pane component
- create a new, DOM based sharing dialog
- move dialog related code to `dialogs.dart`
- the sharing button is always enabled now
- clicking on it, with an unmodified gist, will show you the links for that gist
- clicking on it, with a modified gist, will show ask if you want to share publicly. if so, it'll save to gist.github.com and show you the links created for that gist

@kasperpeulen @lukechurch @sethladd 

This does not include any behavior to re-write the current URL to an '/unsaved' url. The mechanics of that were a bit hard. Perhaps instead we make it clearer when your gist is in an unsaved state? Change something about the UI?

![screen shot 2015-04-14 at 11 48 03 am](https://cloud.githubusercontent.com/assets/1269969/7144847/f9c24bb0-e29c-11e4-9eb2-8d94c1bb3ab8.png)

![screen shot 2015-04-14 at 11 48 11 am](https://cloud.githubusercontent.com/assets/1269969/7144848/01af2a64-e29d-11e4-81c0-9689915d99a3.png)

kasperpeulen commented on lib/dialogs.dart:
```diff
@@ -0,0 +1,105 @@
+      content.add(_div);
+      _padUrl.value = gist.html_url;
+      _gistUrl.value = 'https://dartpad.dartlang.org/${gist.id}';
+
```
_gistUrl and _padUrl are reversed

devoncarew replied:
oops, fixed

kasperpeulen commented on web/index.css:
```diff
@@ -416,3 +380,31 @@ div.toggle {
+  top: 100px;
+  left: 100px;
+  right: 100px;
+}
```
The sharing-dialog feels a bit too wide for me.
Making it 500-600px pixels wide feels better to me:

```
.sharing-dialog {
  top: 100px;
  left: 0;
  right: 0;
  margin: 0 auto;
  width: 600px;
}
```

Or maybe we could use something like this as the default for any dialog ?

kasperpeulen replied:
On your screenshot it doesn't look that, it depends on your screen I guess. On my big widescreen, it looks extremly wide.

devoncarew replied:
agree it's too wide; I like your style changes. I'll make this the default for dialogs.

kasperpeulen commented on lib/dialogs.dart:
```diff
@@ -0,0 +1,105 @@
+    _shareButton.onClick.listen((_) => _performShare());
+
+    // Already sharing.
+    _closeButton = new DButton.button(text: 'Close', classes: 'default');
```
I would like having a close button (in the right top) standard for any dialog.
Something like this maybe ? http://materialdesignicons.com/icon/close

devoncarew replied:
I like it. I may hold off on doing that for this PR however, I already have a lot of changes in it.

kasperpeulen commented on lib/elements/elements.css:
```diff
@@ -196,34 +196,108 @@
+.dialog .title {
+  font-size: 16pt;
+  white-space: nowrap;
+}
```
What about a border bottom like this ?
![](http://imgur.com/mCNJFDc)

css I used:
```
.dialog .title {
  font-size: 16pt;
  white-space: nowrap;
  border-bottom: 1px solid #e5e5e5;
  padding: 5px 10px 10px 10px;
  margin: 0 -10px;
}
```

devoncarew replied:
I had a bottom border on the title area for a while. I wasn't 100% sold on it. I may play around with the style you have here, and try and call back to the 4px border below the page title.

kasperpeulen commented:
Looks like a sweet improvement of the UI :)

I'm trying to make the setting dialog with a similar look.

sethladd commented:
I love the direction.

If the user changes the contents of a shared gist, we do need to make it obvious they have "forked" and are no longer connected to the shared/static version they started with.

Some editors display some small marker when your file is in an unsaved state.

devoncarew commented:
Updated:

- the dialog now has a better max-width
- changed the warning dialog when discarding gists to use the new dialog UI (from the browser default confirm dialog)
- added a dirty bullet to the left of the gist title when the gist is modified

![screen shot 2015-04-14 at 1 57 55 pm](https://cloud.githubusercontent.com/assets/1269969/7147217/2d2ade48-e2ae-11e4-8ead-4a8ba9e18eb1.png)

devoncarew commented:
Merging -
